packages/uri-template/src/compile.ts

```diff
@@ -0,0 +1,91 @@
+import type { TemplateAst } from "./ast.ts";
+import { expand, type Vars } from "./expand.ts";
+import { type EncodingPolicy, match, type MatchOptions } from "./match.ts";
+import { parse } from "./parser.ts";
+
+/**
+ * Options that control how a compiled template behaves during matching.
+ *
+ * ### encoding
+ * Determines how percent-encoded sequences are handled.
+ *
+ * ### strict
+ * If true (default), malformed percent triplets (e.g. "%GZ" or lone "%")
+ * cause matching to fail immediately.
+ * Disabling strict mode may allow more lenient parsing but can lead to ambiguity.
+ */
+export interface CompileOptions {
+  encoding?: EncodingPolicy;
+  strict?: boolean;
+}
+
+/**
+ * A compiled URI template that can efficiently expand and match URLs.
+ *
+ * @typeParam V - The type of variables expected by this template
+ *
+ * @example
+ * ```typescript
+ * const t = compile("{+path}/here");
+ * const url = t.expand({ path: "/foo/bar" }); // "/foo/bar/here"
+ * const match = t.match("/foo/bar/here"); // { vars: { path: "/foo/bar" } }
+ * ```
+ *
+ * @example
+ * ```typescript
+ * const t = compile("/repos{/owner,repo}{?q,lang}");
+  * const url = t.expand({ owner: "alice", repo: "hello/world", q: "a b", lang: "en" });
```

Comment:
The lint is failing due to this line:

```suggestion
 * const url = t.expand({ owner: "alice", repo: "hello/world", q: "a b", lang: "en" });
```